contentcuration/contentcuration/view/zip_views.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+import time
+import mimetypes
+import os
+import re
+import zipfile
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.http import Http404, HttpResponse
+from django.http.response import FileResponse, HttpResponseNotModified
+from django.utils.http import http_date
+from django.views.generic.base import View
+from le_utils.constants import exercises
+from contentcuration.models import generate_storage_url
+
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin
+
+
+# valid storage filenames consist of 32-char hex plus a file extension
+VALID_STORAGE_FILENAME = re.compile("[0-9a-f]{32}(-data)?\.[0-9a-z]+")
+
+# set of file extensions that should be considered zip files and allow access to internal files
+POSSIBLE_ZIPPED_FILE_EXTENSIONS = set([".perseus", ".zip", ".epub", ".epub3"])
+
+
+# DISK PATHS
+
+def get_content_folder_path(datafolder):
+    return os.path.join(
+        datafolder,
+        "content",
+    )
+
+def get_content_storage_folder_path(datafolder=None):
+    return os.path.join(
+        get_content_folder_path(datafolder),
+        "storage",
+    ) if datafolder else settings.STORAGE_ROOT
+
+def get_content_storage_file_path(filename, datafolder=None):
+    assert VALID_STORAGE_FILENAME.match(filename), "'{}' is not a valid content storage filename".format(filename)
+    return os.path.join(
+        get_content_storage_folder_path(datafolder),
+        filename[0],
+        filename[1],
+        filename,
+    )
+
+class ZipContentView(View):
+
+    def get(self, request, zipped_filename, embedded_filepath):
+        """
+        Handles GET requests and serves a static file from within the zip file.
+        """
+        assert VALID_STORAGE_FILENAME.match(zipped_filename), "'{}' is not a valid content storage filename".format(filename)
+
+        # calculate the local file path to the zip file
+        zipped_path = generate_storage_url(zipped_filename).lstrip('/')
```

Comment:
Since the storage_url is relative, the `/` at the beginning was preventing the zipcontent endpoint from finding the correct location of the zip file